tap_snowflake/client.py

```diff
@@ -76,12 +102,131 @@ def discover_catalog_entries(self) -> list[dict]:
+        implementation.
+
+        Any stats not requested but available 'for free' while collecting other stats
+        may be included in the returned profile. For example, the base implementation
+        cannot calculate COLUMN_NULL_VALUES without also calculating TABLE_ROW_COUNT and
+        COLUMN_NONNULL_VALUES. The additional stats therefor be returned along with the
+        requested stats. The consumer should ignore or disregard any stats not needed.
+
+        Note: Gathering stats can take a long time. The implementation should attempt
+        to combine stats gathering into fewer tables scans where possible and only
+        spend time pulling in requested stats.
+
+        Returns:
+            A TableProfile object. Stats may be left null if not requested, not
+            available, or not implemented. Callers should check each field for null
+            values and handle null as 'not available'.
+        """
+        profile_columns = profile_columns or []
+        expressions: list[str] = []
+        if (
+            ProfileStats.TABLE_ROW_COUNT in stats
+            or ProfileStats.COLUMN_NULL_VALUES in stats
+        ):
+            expressions.append("count(1) as row_count")
+        if ProfileStats.TABLE_SIZE_IN_MB in stats:
+            self.logger.debug(
+                "TABLE_SIZE_IN_MB stats not implemented for this provider."
+            )
+        for col in profile_columns or []:
+            if ProfileStats.COLUMN_MIN_VALUE in stats:
+                expressions.append(f"min({col}) as min__{col}")
+            if ProfileStats.COLUMN_MAX_VALUE in stats:
+                expressions.append(f"max({col}) as max__{col}")
+            if (
+                ProfileStats.COLUMN_NONNULL_VALUES in stats
+                or ProfileStats.COLUMN_NULL_VALUES in stats
+            ):
+                expressions.append(f"count({col}) as nonnull__{col}")
+            if ProfileStats.COLUMN_NULL_VALUES in stats:
+                expressions.append(f"count(1) - count({col}) as null__{col}")
+        result_dict = self.connection.scalars(
+            text(f"SELECT {', '.join(expressions)}) FROM {full_table_name})")
+        )
+        return TableProfile(
+            row_count=result_dict.get("row_count", None),
+            size_in_mb=result_dict.get("size_in_mb", None),
+            column_profiles={
+                col: ColumnProfile(
+                    min_value=result_dict.get(f"min__{col}", None),
+                    max_value=result_dict.get(f"max__{col}", None),
+                    null_values=result_dict.get(f"null__{col}", None),
+                    nonnull_values=result_dict.get(f"nonnull__{col}", None),
+                )
+                for col in profile_columns
+            },
+        )
+
 
 class SnowflakeStream(SQLStream):
     """Stream class for Snowflake streams."""
 
     connector_class = SnowflakeConnector
 
+    def _sync_batches(
```

Comment:
I'm not sure what the best way to test this functionality would be. If we have a recording of expected STDOUT, that might be a good way to go.

Reply:
Yeah, having a snapshot of the output from a well-defined table (e.g. seeded via CSV) would help here.